dace/sdfg/infer_types.py

```diff
@@ -381,6 +382,8 @@ def _get_storage_from_parent(data_name: str, sdfg: SDFG) -> dtypes.StorageType:
     parent_sdfg = parent_state.parent
 
     # Find data descriptor in parent SDFG
+    # NOTE: Assuming that all members of a Structure have the same storage type.
+    data_name = data_name.split('.')[0]
```

Comment:
To clarify, this is a developer note and doesn't need to propagate to the user. Currently, the structure creation routines do not ensure common storage type, lifetime, etc. I am suggesting that this should be the case and, for the time being, this is enforced by ignoring the storage type, lifetime, etc., of the nested data. I wouldn't put a warning here; rather make it clear in the documentation (to be written when the nested data design starts to finalize).

Reply:
Also, this may change. I discussed it a bit with Tal and didn't see any legitimate case where part of the structure should have a different storage type or lifetime than the rest, but maybe we have missed something.